### ISSUE TITLE
Setting path to OCaml compiler in configuration file

### ISSUE THREAD

**Issue by nojb** [ocaml/dune#1692]
Currently `dune` expects to find the OCaml compiler in the `PATH`. This is inconvenient when dealing with multiple compilers simultaneously, since it requires setting the `PATH` between each run.

What do you think about allowing to set the path to the OCaml compiler in some configuration file?

**Comment by diml** [comment]
That seems fine to me. That could help for #398 as well.

**Comment by rgrinberg** [comment]
> What do you think about allowing to set the path to the OCaml compiler in some configuration file?

How about allowing to override any variables (including those in ocaml-config) in the workspace file. E.g:

```
(context
 (opam
  (switch 4.06.1)
  (vars
   ocamlopt /foo/bar/ocamlopt)))
```

**Comment by nojb** [comment]
Thinking about this a bit more, I think the simplest would be to have a further case `(context (bin PATH))` where `PATH` would point to the `bin` directory of an OCaml installation. `dune` would then look for the compiler tools inside `PATH` and take it from there.

Overriding individual variables from `ocaml-config` may be useful in other situations but it does not seem principled enough if you just want to choose a different compiler toolchain.

**Comment by nojb** [comment]
Actually maybe we can combine both proposals: we could have a context type
```
(context (explicit
  (ocamlc ...)
  (findlib_path ...)
  ...))
```
where we can override individual variables and otherwise deduce them from those given. The only variable absolutely required is `ocamlc`.

**Comment by nojb** [comment]
I ended up going for something much simpler, see #2426

**Comment by nojb** [comment]
Fixed in #2426
